src/Microsoft.Health.Fhir.Core/Models/OperationOutcomeIssue.cs

```diff
@@ -16,17 +17,31 @@ public class OperationOutcomeIssue
              string diagnostics = null,
              CodableConceptInfo detailsCodes = null,
              string detailsText = null,
-             string[] location = null)
+             string[] expression = null)
         {
             EnsureArg.IsNotNullOrEmpty(severity, nameof(severity));
             EnsureArg.IsNotNullOrEmpty(code, nameof(code));
 
             Severity = severity;
             Code = code;
             DetailsCodes = detailsCodes;
+
+            if (expression != null)
+            {
+                if (DetailsCodes == null)
+                {
+                    DetailsCodes = new CodableConceptInfo();
+                }
+
+                DetailsCodes.Coding.Add(new Hl7.Fhir.Model.Coding(Resources.Meta, Resources.Meta, Resources.OperationOutcomeLocationDeprication));
```

Comment:
I haven't seen this field used this way before, but it was the only place I could find that made sense to add it. @CaitlinV39 what are your thoughts? When this field is eventually removed it will be a breaking change.

Reply:
@LTA-Thinking - is this warning to users who are actively working with `Location` or more of an informational about the `OperationOutcome` resource that is being returned?

Keep in mind that `Location` and `Expression` should not be the same string. `Expression` uses FHIRPath while `Location` [uses XPath and some hotkey logic](https://chat.fhir.org/#narrow/stream/179166-implementers/topic/OperationOutcome.20location.20vs.2E.20expression). Location is actually XML specific - so I wonder if we ever were returning the correct value here. Do you have any data samples comparing the two?

If we did correctly return `Location` previously, then returning both is the right behavior. `Deprecated` means we should keep the current behavior until it is flagged as withdrawn. If we know the user is explicitly using this field a warning is great. If we don't know this, then I worry the warning would just be noise as deprecated fields are a normal part of the HL7 lifecycle.

- [Sample 1 of Location and Expression](https://hl7.org/fhir/operationoutcome-example.json.html)
- [Sample 2 of Location and Expression](https://hl7.org/fhir/operationoutcome-example-validationfail.json.html)

Reply:
@mikaelweave 
We currently use FHIR path for `Location` and have for as long as I've been aware of. Here is an example:
```
{
    "severity": "error",
    "code": "invalid",
    "diagnostics": "Element with min. cardinality 1 cannot be null",
    "location": [
        "Observation.StatusElement"
    ]
},
```
From looking at the spec this is a bug, but as the field is now deprecated and we haven't had any customers complain about it I don't think it would be worth fixing.
The reason I added the warning is because we don't know whether the user is using the `Location` field or not, which is why I would also be against just dropping it. I don't like introducing breaking changes if it can be avoided. 
Given the incorrect usage of `Location` my thought would be to have both fields for a few releases and then drop it. We don't need to wait for the FHIR spec to obsolete the field. This would get us in line with the FHIR spec and give our customers some warning of the change.

Thoughts?

Reply:
@LTA-Thinking 
Would the example now look like below with the code in the PR? Or am I missing something in the logic? 😅 

If the new output looks something like below, my only concern is we have the possibility of mixing error issues inside of a single issue. The `issue` element inside of `OperationOutcome` is defined as "A single issues associated with the action". It the output would result in two separate errors/warnings inside a single `issue` it doesn't seem right.

Also is it possible to have two codings inside of `OperationOutcome.issue.details.coding` that have different meanings?

As for leaving both `location` and `expression`, I agree with your thoughts. We don't want to break anyone currently using `location` even if it's not XPath! Great point there. 

```json
{
  "resourceType": "OperationOutcome",
  "id": "101",
  "issue": [
    {
      "severity": "error",
      "code": "invalid",
      "details": {
        "coding": [
            {
                "system": "???",
                "code": "OperationOutcome.Location is deprecated, please use OperationOutcome.Expression"
            }
        ]
      },
      "diagnostics": "Element with min. cardinality 1 cannot be null",
      "location": [
        "Observation.StatusElement"
      ],
      "expression": [
        "Observation.StatusElement"
      ]
    }
  ]
}
```

Reply:
@LTA-Thinking 

How about this? Assuming `OperationOutcome.Location` is FHIRPath, [comments](https://hl7.org/fhirpath/#comments) are valid in the path. While it is outside the box, it fixes the complaint about the previous approach being slightly out of spec.

```json
{
    "resourceType": "OperationOutcome",
    "id": "f337c346-f907-457a-8910-2b3bb4a8efe4",
    "meta": {
        "lastUpdated": "2022-04-06T14:00:00.448395+00:00"
    },
    "issue": [
        {
            "severity": "error",
            "code": "invalid",
            "diagnostics": "Element with min. cardinality 1 cannot be null",
            "location": [
                "Observation.StatusElement // OperationOutcome.Location is deprecated, please use OperationOutcome.Expression"
            ],
            "expression": [
                "Observation.StatusElement"
            ]
        }
    ]
}
```

Reply:
Here is how it looks after the change: 
```
{
    "resourceType": "OperationOutcome",
    "id": "e46fd007-9034-4a81-b01a-4a2e3d16d1d6",
    "meta": {
        "lastUpdated": "2022-04-07T21:58:47.0764885+00:00"
    },
    "issue": [
        {
            "severity": "error",
            "code": "invalid",
            "diagnostics": "Element with min. cardinality 1 cannot be null",
            "location": [
                "Observation.StatusElement // OperationOutcome.Location is deprecated, please use OperationOutcome.Expression"
            ],
            "expression": [
                "Observation.StatusElement"
            ]
        }
    ]
}
```